File: BambooBank-Spring-Java/src/main/java/BambooBank_Spring_Java/pojos/Employee.java
package BambooBank_Spring_Java.pojos;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;
import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Component
@Scope("prototype")
public class Employee {

    @Id @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "EMP_ID_GEN")
    @SequenceGenerator(name = "EMP_ID_GEN", sequenceName = "EMP_ID_SEQ")
    private Long id;

    @Column(name = "firstname")
    private String firstName;

    @Column(name = "lastname")
    private String lastName;

    private String email;

    private String address;

    private BigDecimal salary;

    @Column(name = "hiredate")
    private LocalDate hireDate;

    // Constructor
    public Employee(String firstName, String lastName, String email, String address, BigDecimal salary, LocalDate hireDate) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.address = address;
        this.salary = salary;
        this.hireDate = hireDate;
    }

    // Default constructor (for JPA)
    public Employee() {
    }

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public BigDecimal getSalary() {
        return salary;
    }

    public void setSalary(BigDecimal salary) {
        this.salary = salary;
    }

    public LocalDate getHireDate() {
        return hireDate;
    }

    public void setHireDate(LocalDate hireDate) {
        this.hireDate = hireDate;
    }
}

File: BambooBank-Spring-Kotlin/src/main/kotlin/bamboobank_spring/bamboobankspringkotlin/pojos/Employee.kt
package bamboobank_spring.bamboobankspringkotlin.pojos

import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import java.math.BigDecimal
import java.time.LocalDate
import jakarta.persistence.*

@Entity
@Component
@Scope("prototype")
data class Employee(
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "EMP_ID_GEN")
    @SequenceGenerator(name = "EMP_ID_GEN", sequenceName = "EMP_ID_SEQ")
    val id: Long = 0,

    @Column(name = "firstname")
    var firstName: String,

    @Column(name = "lastname")
    var lastName: String,

    var email: String,

    var address: String,

    var salary: BigDecimal,

    @Column(name = "hiredate")
    var hireDate: LocalDate
)